src/components/contextual/pages/pool/invest/InvestPage.vue

```diff
@@ -0,0 +1,52 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import usePoolTransfers from '@/composables/contextual/pool-transfers/usePoolTransfers';
+import { usePool } from '@/composables/usePool';
+import useBreakpoints from '@/composables/useBreakpoints';
+import InvestPageMyWallet from './InvestPageMyWallet.vue';
+import InvestPageAccordion from './InvestPageAccordion.vue';
+import InvestPageInvestBlock from './InvestPageInvestBlock.vue';
+
+/**
+ * COMPOSABLES
+ */
+const { pool } = usePoolTransfers();
+const { isDeepPool } = usePool(pool);
+const { upToLargeBreakpoint } = useBreakpoints();
+
+/**
+ * COMPUTED
+ */
+const poolSupportsGeneralisedJoin = computed<boolean>(() => {
+  return !!(pool.value && isDeepPool.value);
+});
+</script>
+
+<template>
+  <div class="invest-page-layout-grid">
+    <div v-if="!upToLargeBreakpoint" class="col-span-5">
+      <InvestPageMyWallet
+        :poolSupportsGeneralisedJoin="poolSupportsGeneralisedJoin"
```

Comment:
Not a fan of this prop drilling. Why does this need to be passed down as a prop? All these components should have access to the pool state, so in specific cases, it can just check if the pool isDeep.